lib/src/tag.dart

```diff
@@ -32,10 +33,24 @@ class Tag {
       lastLine = slr.readNextLine()!;
     }
 
-    final objectSha = headers['object']!.single;
-    final type = headers['type']!.single;
-    final tag = headers['tag']!.single;
-    final tagger = headers['tagger']!.single;
+    String objectSha;
+    String type;
+    String tag;
+    String tagger;
+
+    if (headers['object'] != null) {
+      // Annotated Tag
+      objectSha = headers['object']!.single;
+      type = headers['type']!.single;
+      tag = headers['tag']!.single;
+      tagger = headers['tagger']!.single;
+    } else {
+      // Lightweight Tag
+      objectSha = ref.sha;
+      type = 'lightweight';
+      tag = ref.reference.split('/').last;
```

Comment:
I think this should use a regex. And track the first bits stictly.

Reply:
Just to clarify, you want the `split` to use a RegEx instead? Or to use something, like allMatches?

Is there a test case you are concerned about that I could implement so I know when I found a solution? RegEx is not my strong suit.

Reply:
@kevmoo, I need more direction on what you want regarding the regex pattern (what's the targeted matches?) and clarity on what you mean by tracking first bits.

Reply:
I just pushed a commit!